airflow/models/taskinstance.py

```diff
@@ -1311,12 +1311,6 @@ def _run_finished_callback(self, error: Optional[Union[str, Exception]] = None)
             if task.on_success_callback is not None:
                 context = self.get_template_context()
                 task.on_success_callback(context)
-        elif self.state == State.UP_FOR_RETRY:
-            task = self.task
-            if task.on_retry_callback is not None:
-                context = self.get_template_context()
-                context["exception"] = error
-                task.on_retry_callback(context)
```

Comment:
I have moved this to handle_failure method since UP_FOR_RETRY is not a finished callback and that makes it not to be called when on_retry_callback is set